### PR TITLE
Tuple elim

### PR DESCRIPTION
**Pull request by mbudiu-vmw**
This is paired with https://github.com/p4lang/p4-spec/pull/877
Note that tuple fields are not (yet) left values.

### CODE REVIEW

File: frontends/common/constantFolding.cpp [hunk 1]
```diff
@@ -580,31 +580,42 @@ const IR::Node* DoConstantFolding::postorder(IR::Member* e) {
...
-            BUG_CHECK(ne != nullptr, "Could not find field %1% in initializer %2%", e->member, si);
-            return CloneConstants::clone(ne->expression);
+            if (auto list = expr->to<IR::ListExpression>()) {
+                result = CloneConstants::clone(list->components.at(static_cast<size_t>(index)));
+            }
         } else {
-            BUG("Unexpected initializer: %1%", expr);
+            auto structType = type->to<IR::Type_StructLike>();
+            if (structType == nullptr)
+                BUG("Expected a struct type, got %1%", type);
```
Comment (ChrisDodd) [inline frontends/common/constantFolding.cpp:594]: probably should be

    } else if (auto structType = type->to<IR::Type_StructLike>()) {
            :
    } else {
        BUG("Expected a struct or tuple type, got %1%", type);

**Review by ChrisDodd — APPROVED** [review]

**Comment by mbudiu-vmw** [comment]
@fruffy : this fails in the gauntlet check, but the diagnostic is not clear. How do we interpret this?

**Comment by mbudiu-vmw** [comment]
I actually suspect that what happened is that this is introducing a new feature in the language (tuple fields), which it not yet modeled in Gauntlet, so Gauntlet fails.

**Comment by fruffy** [comment]
Ugh. I knew this would happen eventually. Yes this construct is not supported yet. I think I should let the check only fail if it reports an actual validation error. Otherwise this will keep happening. Implemented a hotfix that should pass this test at least.

**Comment by fruffy** [comment]
Actually the fix for this was really simple. It was just a field naming issue.

**Pull request merged**